### ISSUE TITLE
zune-png panics on decoding some valid inputs

### ISSUE THREAD

**Issue by Shnatsel** [etemesi254/zune-image#23]
Run the fuzz target added in https://github.com/etemesi254/zune-image/pull/22 : `cargo +nightly fuzz run roundtrip`

It finds the following panic in a few seconds:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Corrupt data
', /home/shnatsel/Code/zune-image/zune-png/src/decoder.rs:264:50
```

**Comment by etemesi254** [comment]
Could you confirm now?

Should have been the zlib bug which was fixed by 2f55ce28f305630b97d98d36bef429e99a789958

On my side it's running been for some good minutes without failing

**Comment by Shnatsel** [comment]
Yes, that's probably it. I'll run it some more just to be sure, but I'll go ahead and close this.
